src/Orleans/Messaging/GatewayConnection.cs

```diff
@@ -189,7 +189,7 @@ public void Connect()
         protected override bool PrepareMessageForSend(Message msg)
         {
             // Check to make sure we're not stopped
-            if (Cts.IsCancellationRequested)
+            if (Cts!=null && Cts.IsCancellationRequested)
```

Comment:
This is not doing the right thing now I believe.
If it's null, it should probably just return, because it's past anything it can do. But at the very least, if it's null, it should treat as canceled, not as not canceled as it's doing now.